lib/convection/control/stack.rb

```diff
@@ -586,12 +582,39 @@ def get_events(pages = nil, stack_name = id)
         end
       rescue Aws::CloudFormation::Errors::ValidationError # Stack does not exist
       rescue Aws::EC2::Errors::RequestExpired
-        raise unless @credential_error_count <= @credential_error_max_retries
+        raise if reached_max_credential_retries?
+
+        wait_for_credential_refresh
+
+        reload_cf_client
+        retry
+      end
+
+      def client_options
+        client_options = { region: @region }
+        client_options[:credentials] = @credentials unless @credentials.nil?
 
+        client_options
+      end
+
+      def reload_cf_client
+        Aws.shared_config.fresh if @credentials.nil?
```

Comment:
Here be dragons...  `Aws.shared_config`[1] not part of the public api.

[1] http://www.rubydoc.info/gems/aws-sdk-core/2.5.1/Aws.shared_config